source/common/http/http1/codec_impl.h

```diff
@@ -238,31 +238,16 @@ class ConnectionImpl : public virtual Connection,
 
 protected:
   ConnectionImpl(Network::Connection& connection, CodecStats& stats, const Http1Settings& settings,
-                 http_parser_type type, uint32_t max_headers_kb, const uint32_t max_headers_count,
+                 MessageType type, uint32_t max_headers_kb, const uint32_t max_headers_count,
                  HeaderKeyFormatterPtr&& header_key_formatter);
 
-  // The following define special return values for http_parser callbacks. See:
-  // https://github.com/nodejs/http-parser/blob/5c5b3ac62662736de9e71640a8dc16da45b32503/http_parser.h#L72
-  // These codes do not overlap with standard HTTP Status codes. They are only used for user
-  // callbacks.
-  enum class HttpParserCode {
-    // Callbacks other than on_headers_complete should return a non-zero int to indicate an error
-    // and
-    // halt execution.
-    Error = -1,
-    Success = 0,
-    // Returning '1' from on_headers_complete will tell http_parser that it should not expect a
-    // body.
-    NoBody = 1,
-    // Returning '2' from on_headers_complete will tell http_parser that it should not expect a body
-    // nor any further data on the connection.
-    NoBodyData = 2,
-  };
-  int setAndCheckCallbackStatus(Status&& status);
-  int setAndCheckCallbackStatusOr(Envoy::StatusOr<HttpParserCode>&& statusor);
+  int setAndCheckCallbackStatus(Http::Status&& status) override;
+  int setAndCheckCallbackStatusOr(Envoy::StatusOr<ParserStatus>&& statusor) override;
 
   bool resetStreamCalled() { return reset_stream_called_; }
-  Status onMessageBeginBase();
+
+  Status onMessageBegin() override;
```

Comment:
comment on the difference between these two?

Reply:
Added a comment. `onMessageBeginStatus()` is the connection level method